src/base/SanityChecker.ml

```diff
@@ -482,18 +487,403 @@ struct
           stmt_iter component.comp_body)
   end
 
+  (* ************************************************** *)
+  (* ******** Check unboxing of option results ******** *)
+  (* ************************************************** *)
+
+  module CheckUnboxing = struct
+    module SCIdentifierComp = struct
+      include SCIdentifier.Name
+      include Comparable.Make (SCIdentifier.Name)
+    end
+
+    module SCIdentifierSet = Set.Make (SCIdentifierComp)
+
+    let emp_ids_map = Map.empty (module SCIdentifierComp)
+    let emp_ids_set = SCIdentifierSet.empty
+
+    let is_option_name id =
+      String.equal "Option" @@ SCIdentifier.Name.as_string (get_id id)
+
+    let is_option_ty id =
+      let re = Str.regexp "Option.*$" in
+      Str.string_match re (SType.pp_typ (ER.get_type (get_rep id)).tp) 0
```

Comment:
I'd suggest to solve with pattern-matching and move this predicate somewhere here: https://github.com/Zilliqa/scilla/blob/master/src/base/Type.ml#L479